src/components/structures/TimelinePanel.tsx

```diff
@@ -1637,7 +1637,9 @@ class TimelinePanel extends React.Component<IProps, IState> {
         let i = events.length - 1;
         let userMembership = "leave";
         for (; i >= 0; i--) {
-            const timeline = room.getTimelineForEvent(events[i].getId());
+            const timeline = isThreadTimeline
+                ? room.getTimelineForEvent(events[i].getId()!)
+                : room.getUnfilteredTimelineSet().getTimelineForEvent(events[i].getId()!);
```

Comment:
I'm afraid I'm not really following why this is a sensible change, or why it fixes the problem. Given that `room.getTimelineForEvent` falls back to `room.getUnfilteredTimelineSet().getTimelineForEvent` for events that are not in a thread, why is it not a no-op?

Some comments explaining this would be appreciated.

Reply:
Fair comment! I've actually managed to simplify the change and use the timeline set that has been passed from props. 
Also added some regression tests.